stubs/qrcode/qrcode/util.pyi

```diff
@@ -25,21 +30,26 @@ def BCH_type_info(data): ...
 def BCH_type_number(data): ...
 def BCH_digit(data): ...
 def pattern_position(version): ...
-def mask_func(pattern): ...
+def mask_func(pattern: _MaskPattern) -> Callable[[int, int], bool]: ...
 def mode_sizes_for_version(version): ...
 def length_in_bits(mode, version): ...
 def check_version(version) -> None: ...
 def lost_point(modules): ...
-def optimal_data_chunks(data, minimum: int = 4) -> Generator[Incomplete, None, None]: ...
-def to_bytestring(data): ...
-def optimal_mode(data): ...
+def optimal_data_chunks(data: str | bytes, minimum: int = 4) -> Generator[QRData, None, None]: ...
+def to_bytestring(data: str | bytes) -> bytes: ...
```

Comment:
In theory this (and all the `data` arguments that ultimately call this) accept arbitrary objects, since this calls `str()` on `data`. Still, this is quite unsafe, and I feel calling `str()` manually is not too much of a burden. (And we can always change the stubs if users have valid use cases.)